### ISSUE TITLE
(net) exports in the graph "Origin of electricity in the last 24 hours"

### ISSUE THREAD

**Issue by ggaulier** [tmrowco/electricitymap#1030]
Congratulation first
A suggestion: put exports or net exports (in negative) in the graph "Origin of electricity in the last 24 hours"
Would like to know how it correlates with renewables production in Germany.
Best

**Comment by alixunderplatz** [comment]
Hi @ggaulier,
do you mean [something like this from energy-charts.de](https://energy-charts.de/power_de.htm?source=conventional&year=2018&week=1):

![image](https://user-images.githubusercontent.com/25743609/35471251-b8943958-0357-11e8-88fa-583f5ae86d48.png)

It would definitely help to "flat out" the "origin of electricity" chart a bit, so this will also show the actual "demand profile".
I noticed that Estonia and Latvia, for example, are often a mess when it comes to the origin, because they produce and import electricity that is meant to supply Lithuania, too. That causes very "artistic" origin-graphs.

![image](https://user-images.githubusercontent.com/25743609/35471322-370621ba-0359-11e8-984d-b0c075fcab55.png)

If you displayed all the exports as a line in the negative area below the axis and the difference of "origin minus exports" as line for "domestic demand" this could help a bit

**Comment by corradio** [comment]
One thing to keep in mind is that exports do not change the consumption mix, which is why we don't show them.
Also, we would have to change the title, because it doesn't make sense to have negative quantities in the "origin of electricity"

**Comment by alixunderplatz** [comment]
Right, this is not supposed to impact the "consumption mix" / the "CO2 intensity" of a country/region.

I know you are aware of what this looks like, but I will try to outline the effects and intention of including exports in the graph using Estonia as an example:
Looking at the second screenshot I posted above from Estonia, it might be confusing to some people, why such strange peaks from power import appear. Looking at the exchange arrows on the map may make things a bit clearer already, no doubt. 

Balancing and displaying the exchanges with domestic generation and demand in a seperate graph might further improve the comphrehensibility when it comes to that. I know, adding an additional fourth graph can be "overloading" the left panel optically and is not ideal.

I tried to highlight what including exports in the negative part will look like.
In this special case, there is a clear correlation between exports and imports, since almost everything that comes from Finland is exclusively related to market effects in Lithuania and Latvia. There is a generation deficit in Latvia and, even worse, Lithuania. So energy is partially imported from Finland, via DC cables, which causes a power flow towards the southern Balltic states Latvia and Lithuania via Estonia and Russia/Belorussia (Baltic states are still a part of that synchronous region). Lucky Estonia gets its CO2 intensity "improved", because the domestic energy generation is being "pushed" to the south through the grid. No problem, this is what the algorithm is supposed to do for such reasons.

![image](https://user-images.githubusercontent.com/25743609/35473399-44b81bf2-0380-11e8-9c31-15a2ab72763d.png)

Including exports in the negative part as an area will mainly be another optical support.

Having a simplified graph showing the calculated balanced load (or given load from the data) in a state will show how much "unrequested" generation and import is flowing over the borders.
In this second diagram, you can clearly see that the balance (green) between exports (blue) and imports (yellow) is close to 0 MW most of the time for Estonia, and the calculated domestic demand (red) shows a typical load profile, which is almost entirely covered by domestic generation (black).

![image](https://user-images.githubusercontent.com/25743609/35474649-a9f09cc2-0391-11e8-9b8d-8a143f728327.png)

Such a simplified graph can have a similar outcome for countries that are highly interconnected and have a relatively low demand compared to their transmission capacity, like Switzerland, Austria, for example. These countries actually don't deserve their CO2 intensity to be spoiled by "external" sources, but that's how it is.
For other regions with a generation deficit, it will show the gap between demand and domestic generation, which is the cause for physical imports.

**Comment by HansHyde** [comment]
Alex, what you are seeing in your second graph is what I have been trying to describe to all... to no avail.

You call it an algorithm, I've called it a calculation, formula or equation. But regardless what it is called, the internal country calculations are wrong. And when export/imports are included and cause 'peaks' and large variations from the 'carbon intensity', it is not the import/exports that are wrong, it is that the internal country parser calculations are wrong and bias themselves.

If anyone were to write down the algorithms, define each of their components, then cancel out their labels/units they would see the fundamental equation is wrong.

The only reason EM's 'Carbon Intensity' is always somewhere between the lowest EM setpoint, eg Battery = 0 and Coal = 820, is because the equation(s) is wrong.

TBH, I have turned towards creating a report & an Abstract to submit to scientific journals, such as IEEE, to document this... major fail.

But as it is such a 'major fail', it is difficult to visually display/chart in a way that is comprehensible to an outsider. When corrected values can be off by over 1000%, one can't even see/distinguish on the chart the the values being measured in the first place.

I don't know who came up with the fundamental equation, Oli or Bruno, but it does not pass scientific rigor and at present, only 'passes' EM's guidelines. The EM 'Carbon Intensity' only proves itself... which 'makes sense' as it is running its own equation. But it does not pass scientific methodology, nor basic algebra.

**Comment by HansHyde** [comment]
The 'origins of electricity' graph, such as in Alberta, is actually correct. Those units/numbers displayed on the graph are actual, yet nowhere with the EM universe do they correlate with the EM 'Carbon Intensity'.

That chart IS the aggregated carbon eq emissions. It is REAL/ACTUAL emissions based on the IPCC-based fuel set points EM is using.

EM's 'Carbon Intensity' is the readout of a series of false/incorrect calculations, that EM alone has determined to be correct. It is not correct.

There are no peer-reviewed reports validating EM's methodology. And every report that is used to supersede the IPCC-based set points EM is using, are not validations of EM's methodology, they are further validations of the IPCC-based ranges for fuel-based medians & ranges.

These are not the same things.

How in the hell, does the equation's first component multiplied by the fuel output divided by the total output equal the fuel's contribution to 'carbon intensity'??? It doesn't, except for the units for fuel output & total output cancel themselves out, resulting in the first component's equaling the final value's units.

And when the units for each of the 3 components & the result are not understood by all, then the mistakes/errors continue on.

The major flaw of EM is not the concept or the display or the contributions of those helping here. It is that all have the expectation that the fundamental calculation/formula/algorithm for 'Carbon Intensity' is correct and that it was validated by neutral parties. But it has not been validated and proven accurate.

**Comment by HansHyde** [comment]
The belief/expectations by all EM contributors that the 'Carbon Intensity' calculation is correct, is masking all the other evidence (when present or presented) to the contrary.

The 'Carbon Intensity', because it is weighted twice, not only weighted 1 time, is the problem with the calculation.

Multiply the fuel set point by the fuel output then divide by the total output only makes logical sense if one is attempting to cancel out units. If all units are taken at face-value, all works out. But face-value is not scientific value.

Dividing by 1000 only moves the decimal place - into an 'expected' position. Dividing by 3600 seconds in an hour gives a true result.

Get rid of the division by total output, divide by 3600, and then 'Carbon Intesity' Rate follows/correlates exactly to what one sees in the 2nd Alberta graph for aggregated emissions.

**Comment by corradio** [comment]
@alixunderplatz you have a valid point.
I will try to think of a good way to show the part of the imports that are solely there for exporting purposes. I think the angle could be to highlight that while importing higher-co2 electricity is a bad thing, it is not easy to avoid as they come from exporting needs/requirements (as opposed to a deficit in production).

Not an easy story to tell visually and simply. It might very well be that adding the negative part is the only option.

**Comment by brunolajoie** [comment]
Open question here: 
I'm wondering how we could ever tell, rigorously, that "a country is importing for the sole purpose of exporting" ? 

If its a bit "unfair" to attribute these imports (and the related CO2 imported) to such country, what would be the alternative? Getting back to "production-only" CO2 intensity would also be unfair for some other countries, that are clearly importing for their own consumption purposes.

**Comment by ggaulier** [comment]
Hello,
Thanks Alex, this is what I mean
I'm not a specialist so I'm not sure I understand well other replies! Interesting anyway.
Best

**Comment by alixunderplatz** [comment]
@brunolajoie 
Hi Bruno! In my comment, I just wanted to point out that it might be helpful to see an "exchange balance" or something like a "domestic demand curve" as an optical comparison to the generation or the individual exchanges.

I have no doubt, that any exports influence another region's CO2 intensity, because the grid is highly interconnected and electricity mostly follows physical laws (in AC grids).

To your question "how could we ever tell that a country is importing for the purpose of exporting", I want to point at the example I gave for Estonia: by balancing load vs. generation. That particular export flow is caused by an energy input via DC cable, which is, in this special case, a 100% market-driven import of the nordpool market to cover the neighbours' demand. In this case, it might be helpful to take a look at the actual "commercial flows" of a country. I predict from my gut feeling, without having looked it up, the commercial flow for Estonia it will mostly be 0 MW for the displayed hours of that day, for Lithuania around 1000 MW import and Latvia a few hundred MW import.

Again, in my opinion, it could be helpful to have a simplified overview of "total generation vs. load vs. imports vs. exports".

**Comment by brunolajoie** [comment]
Good point! Commercial flow might be helpful indeed to attribute "responsibility" of consumption (versus physical flow). But commercial flows are only day-ahead, not rela time...so it won't responsibilize consumers based on the real time status of the grid...It's a tricky choice! Do we want to make a map close to physical reality or commercial plans? but hey interesting thoughts!  Happy to discuss further if we see practical implementations potential to the EMap

**Comment by jarek** [comment]
I was also thinking about this. My suggestion would be to draw a separate line with value of production+imports-exports on the "origin of electricity" chart when exports are more than ~5% of productions+imports.

**Comment by corradio** [comment]
I like having a simple case to think about things. For example here, El Hierro is having a lot of wind, triggering storage units to use pumped storage.
![image](https://user-images.githubusercontent.com/1655848/38096918-5855773a-3374-11e8-8d2e-8632f3677839.png)

One thing I wanted to do is to have the "origin of electricity" curve represent the load curve (as right now it sums up to the total amount of incoming electricity - i.e. production + imports + discharge).
The drawback summing up to the load is that the gust of wind triggering a storage will not really be visible on the graph (as the load more or less stays the same), and that's a shame.

For me this is a good rationale for showcasing "negative areas" (storage + exports). However, the load curve will still not be visible. There is also a challenge in colouring those areas, as exports will all have the same colour (but maybe that's not a problem).

Another solution is to indeed show an area graph that sums to the load curve, and then show another graph that shows the amount of electricity stored or exported, as positive values. The wind gust will then be visible on the storage graph (not on the load, except for the % change). This will enable the axes to become more readable.

Unfortunately I don't think showing a load curve on top of an area graph with positive / negative values will work due to our colours.

**Comment by corradio** [comment]
This is an example with Germany for example:
![image](https://user-images.githubusercontent.com/1655848/38171999-a9264734-35a4-11e8-86b8-0b57efdc61c2.png)

There's a lot of information.

Here's an idea:
We could show an area chart, coloured by the zone's carbon intensity, showing storage and exports over time. This would even highlight how small the Tesla battery is in AUS.

For the load information, I've personally come to the conclusion that it's more important to show overall volume of electricity incoming + outgoing, instead of "netting" and displaying the load.

Do you agree with adding a new graph?

**Comment by jarek** [comment]
To be honest I have started to like the idea of scaling the origin of electricity so that it matches the load curve. Currently the origin of electricity looks un-intuitive in Balkans and Baltics.

The problem with that is defining what such a chart actually is and what to call it, because it's not origin of electricity in MW/GW anymore. "Fuel mix of local load" is the best I can come up with. Then we lose absolute generation values in MW/GW, which might be a good thing or a bad thing depending how you look at it.

How about scaling generation+imports according to local load, _and_ drawing exports/storage as negative on the same chart? Can it use the same units and would the overall height be "correct"? I'm not sure how the math works on this.

Adding another chart is an option but then we have a lot of charts.

Or we could just decide that the charts for highly interconnected regions like the Baltics are unintuitive because the system behaviour there is complicated and we value accuracy over intuitiveness.

(I think the issues of fairness of scoring import-for-export brought up above are largely separate from the chart display.)

**Comment by brunolajoie** [comment]
Its an interesitng idea...especially if we don't show "GW" but "%" in the scale.

**Comment by corradio** [comment]
@jarek I think that if you scale generations+imports according to load,
you've already subtracted exports/storage, so showing them as negative
doesn't really make sense anymore.
Either we show a "net" graph that scales to load, but then the
exports/storage have already been subtracted, or we show each
generation/import and exports/storage components separately (the sum of
both will be the load). The latter could be shown in one graph
(positive/negative) or in two graphs.

**Comment by systemcatch** [comment]
No comment on showing exports, others have better & more detailed ideas. One thing I will say is that if another chart is added things will become even more cramped than they are now.

It's always bugged me that the origin of electricity graph is hidden and requires scrolling to see it when a zone is clicked on, imo this is one if not the most interesting graph in the whole project yet it's hidden and small. I know that contributors can zoom in to display problems/interesting events but honestly I wouldn't rely on most people being able to do that.

I think we should make the left panel expandable as well as collapsible (#1282) then we can display graphs at bigger sizes and not have to worry about space so much anymore. We could even allow a graph to be expanded into it's own tab/window for serious detail! #crazyideas.

**Comment by corradio** [comment]
I've had an extensive chat @brunolajoie about this topic.
We really want to showcase where electricity comes from and be able to have windgusts to show up.
That prohibits showing a curve that sums up to the load.
The next information that makes sense to show is the storage/exports (you want to see that a windgust triggers a storage/export). That would warrant another graph.

We've tried it and negative graphs won't work due to axis values. Also it complicates the UI significantly.

So if there's anything to improve: add a new graph with "outgoing power" (storage/exports).

**Comment by brunolajoie** [comment]
That will be clear indeed:
- a carbon intensity graph
- a "incoming power" graph, that is the basis for computing carbon intensity
- a "outgoing power" graph below, that does not impact carbon intensity
- a price graph

**Comment by jarek** [comment]
For what it's worth, when using the Electricity Map lately I have wished for a fourth graph for "outgoing power" as suggested by Bruno on May 4. So +1 on this for me.

**Comment by corradio** [comment]
Hi all,

Is this still relevant? Does anyone still miss showing storage / exports?
cc @robertahunt 

Olivier

**Comment by Kongkille** [comment]
We are redoing some of the UI for historical view, and things are quite cramped already so I don't think there are any direct actions here.
